### PR TITLE
Visual improvements

### PR DESCRIPTION
**Pull request by pmaas**
Added some style improvements to integrate the color picker a  little bit better into Neos:

**Before / After**
Bildschirmfoto 2019-07-17 um 12 44 57Bildschirmfoto 2019-07-17 um 12 28 28

I needed to create a custom picker based on the sketch color picker, because I wasn't able to style all components by the `style`-prop. But thereby it is also possible to integrate the reset button into the picker.

### CODE REVIEW

**Comment by Sebobo** [comment]
Wow! Thanks!

I'll review and merge it asap

File: Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Alpha.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import React, { Component, PureComponent } from 'react'
+import reactCSS from 'reactcss'
+import * as alpha from 'react-color/lib/helpers/alpha'
+
+import Checkboard from 'react-color/lib/components/common/Checkboard'
+
+export class Alpha extends (PureComponent || Component) {
```
Comment (Sebobo) [inline Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Alpha.js]: Would be good to define the possible proptypes here

File: Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/ColorPicker.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+import { ColorWrap, Saturation, Checkboard } from 'react-color/lib/components/common'
+import Alpha from './Alpha'
+import Hue from './Hue'
+import Fields from './Fields'
+import PresetColors from './PresentColors'
+import {IconButton} from '@neos-project/react-ui-components';
+
+export const ColorPicker = ({ width, rgb, hex, hsv, hsl, onChange, onReset, onSwatchHover,
+  disableAlpha, presetColors, renderers, styles: passedStyles = {}, className = '' }) => {
```
Comment (Sebobo) [inline Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/ColorPicker.js]: Is `disableAlpha` forwarded to the fields? I cannot see where and PHPStorm shows it's unused.

File: Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Fields.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
+/* eslint-disable no-param-reassign */
+
+import React from 'react'
+import reactCSS, { focus } from 'reactcss'
```
Comment (Sebobo) [inline Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Fields.js]: `focus` does not seem to be used

File: Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Hue.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import reactCSS from 'reactcss'
+import * as hue from 'react-color/lib/helpers/hue'
+
+export class Hue extends (PureComponent || Component) {
+	componentWillUnmount() {
+		this.unbindEventListeners()
+	}
+	
+	handleChange = (e) => {
+		const change = hue.calculateChange(e, this.props.direction, this.props.hsl, this.container)
```
Comment (Sebobo) [inline Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Hue.js]: I get an error `Cannot read property 'h' of undefined` here when using the `hsla` mode and selecting a color from the rgb and alpha bars. Selections from the other fields work fine.

File: Resources/Private/Scripts/ColorPickerEditor/src/ColorPickerEditor.js [hunk 5]
```diff
@@ -1,15 +1,15 @@
 import React, {PureComponent} from 'react';
 import PropTypes from 'prop-types';
-import {SketchPicker} from 'react-color';
+import NeosColorPicker from './NeosColorPicker/ColorPicker';
 import {IconButton} from '@neos-project/react-ui-components';
```
Comment (Sebobo) [inline Resources/Private/Scripts/ColorPickerEditor/src/ColorPickerEditor.js]: `IconButton` is not used anymore

**Comment by pmaas** [comment]
Hey @Sebobo, thanks for the review! And sorry for the late reply, I've been on vacation last week, but I try to do the fixes asap.

File: Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Hue.js (hunk 4 above)
Comment (pmaas) [reply Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Hue.js]: Sorry for the bad testing. I added some last minute style changes and didn't check the functionality afterwards.

File: Resources/Private/Scripts/ColorPickerEditor/src/ColorPickerEditor.js (hunk 5 above)
Comment (pmaas) [reply Resources/Private/Scripts/ColorPickerEditor/src/ColorPickerEditor.js]: Is now removed.

File: Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Fields.js (hunk 3 above)
Comment (pmaas) [reply Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Fields.js]: Is now removed.

File: Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/ColorPicker.js (hunk 2 above)
Comment (pmaas) [reply Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/ColorPicker.js]: Maybe we can make this a future feature. Until this I removed the variable.

**Comment by pmaas** [comment]
Hey @Sebobo, I added the `.eslintrc` from Neos.Ui as recommended by you and fixed the hue/alpha slider bugs. Would be awesome if you find some time to check the changes. Thanks!

**Comment by Sebobo** [comment]
Looks good. I fixed some minor details.

Can you define the missing `propTypes` in Alpha & Hue? Then we're done I think :)

File: Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Alpha.js (hunk 1 above)
Comment (pmaas) [reply Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Alpha.js]: Fixed

**Comment by pmaas** [comment]
> Can you define the missing propTypes in Alpha & Hue? Then we're done I think :)
Done.

**Pull request merged**